irc.go

```diff
@@ -0,0 +1,134 @@
+package twitch
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type ircMessage struct {
+	Raw     string
+	Tags    map[string]string
+	Source  ircMessageSource
+	Command string
+	Params  []string
+}
+
+type ircMessageSource struct {
+	Nickname string
+	Username string
+	Host     string
+}
+
+func parseIRCMessage(line string) (*ircMessage, error) {
+	message := ircMessage{
+		Raw:    line,
+		Tags:   make(map[string]string),
+		Params: []string{},
+	}
+
+	split := strings.Split(line, " ")
+	index := 0
+
+	if strings.HasPrefix(split[index], "@") {
```

Comment:
Blank lines are allowed in IRC and should be ignored. Wouldn't this go out of bounds when the input "line" is empty?

Reply:
An empty line should never reach this function, or even `parseMessage`. That is already handled above in `handleLine`.